types/index.d.ts

```diff
@@ -248,45 +252,75 @@ declare module 'react-sweet-state' {
     dependencies: any[];
   };
 
-  function createSelector<ST, PR, SE, T1>(
-    selector1: InputSelector<ST, PR, T1>,
+  function createSelector<ST, PR = void, SE = void, T1 = void>(
+    selectors: [InputSelector<ST, PR, T1>],
     resultFn: (arg1: T1) => SE
   ): OutputSelector<ST, PR, SE>;
-  function createSelector<ST, PR, SE, T1>(
-    selectors: [InputSelector<ST, PR, T1>],
+  function createSelector<ST, PR = void, SE = void, T1 = void>(
+    selector1: InputSelector<ST, PR, T1>,
     resultFn: (arg1: T1) => SE
   ): OutputSelector<ST, PR, SE>;
-  function createSelector<ST, PR, SE, T1, T2>(
+  function createSelector<ST, PR = void, SE = void, T1 = void, T2 = void>(
     selector1: InputSelector<ST, PR, T1>,
     selector2: InputSelector<ST, PR, T2>,
     resultFn: (arg1: T1, arg2: T2) => SE
   ): OutputSelector<ST, PR, SE>;
-  function createSelector<ST, PR, SE, T1, T2>(
+  function createSelector<ST, PR = void, SE = void, T1 = void, T2 = void>(
     selectors: [InputSelector<ST, PR, T1>, InputSelector<ST, PR, T2>],
     resultFn: (arg1: T1, arg2: T2) => SE
   ): OutputSelector<ST, PR, SE>;
-  function createSelector<ST, PR, SE, T1, T2, T3>(
+  function createSelector<
+    ST,
+    PR = void,
```

Comment:
This all is to make `PR` as `void` in case TS is not able to determine the type. It benefits in cases where there is a selector function which is consuming any prop. If we will not do this it will be treated as `unknown` and `TS` will complain to pass an argument (which can be anything) in the caller function. 

This does not effect any other things.